pkg/cluster/cluster.go

```diff
@@ -205,17 +205,18 @@ func (c *Cluster) send(ev *clusterEvent) {
 }
 
 func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
-	needDeleteCluster := true
+	clusterFailed := false
 
 	defer func() {
-		if needDeleteCluster {
-			c.logger.Infof("deleting cluster")
+		if clusterFailed {
+			c.reportFailedStatus()
```

Comment:
also i am not super concerned about this case right now. if reporting fails, then the cluster cannot connect to API server. Controller cannot do anything meaningful either. But we should fix the root problem: a cluster event should never blocks the controller.